diff --git a/node_modules/solid-start/bin.cjs b/node_modules/solid-start/bin.cjs
index eb7de24..2bdf749 100755
--- a/node_modules/solid-start/bin.cjs
+++ b/node_modules/solid-start/bin.cjs
@@ -31,6 +31,24 @@ const findAny = (path, name) => {
   return null;
 };
 
+prog
+    .command("routes").describe("Show all routes in your app")
+    .action(async ({config: configFile, open, port, root, host, inspect}) => {
+      root = root || process.cwd();
+      const config = await resolveConfig({ mode: "production", configFile, root, command: "build" });
+
+      const { Router } = await import("./fs-router/router.js");
+
+      const router = new Router({
+        baseDir: path.posix.join(config.solidOptions.appRoot, config.solidOptions.routesDir),
+        pageExtensions: config.solidOptions.pageExtensions,
+        ignore: config.solidOptions.routesIgnore,
+        cwd: config.solidOptions.root
+      });
+      await router.init();
+
+      console.log(JSON.stringify(router.getFlattenedPageRoutes(), null, 2));
+    })
 prog
   .command("dev")
   .describe("Start a development server")
